packages/react/src/drafts/SelectPanel2/SelectPanel.tsx

```diff
@@ -337,7 +351,35 @@ const Panel: React.FC<SelectPanelProps> = ({
 }
 
 const SelectPanelButton = React.forwardRef<HTMLButtonElement, ButtonProps>((props, anchorRef) => {
-  return <Button ref={anchorRef} {...props} />
+  const inputProps = useFormControlForwardedProps(props)
+  const [labelId, setLabelId] = useState('')
+  useEffect(() => {
+    const label = document.querySelector(`[for='${inputProps.id}']`)
+    if (label) {
+      if (label.id) {
+        setLabelId(label.id)
+      } else {
+        const newLabelId = `${inputProps.id}--select-panel-button-label`
+        label.id = newLabelId
+        setLabelId(newLabelId)
+      }
+    }
+  }, [inputProps.id])
+
+  const punctuationId = `${inputProps.id}--select-panel-button-punctuation`
+  const punctuation = ','
+  if (inputProps.id && labelId) {
+    return (
+      <>
+        <VisuallyHidden id={punctuationId} aria-hidden="true">
```

Comment:
Made this `aria-hidden="true'` so screen readers aren't distracted by this punctuation. Instead, it's only used in the aria-labelledby attribute

Reply:
🤔 I wonder if it would be more straightforward to just build the label ourselves:

```jsx
const labelText = document.getElementById(labelId)?.textContent ?? ""
const buttonText = document.getElementById(inputProps.id)?.textContent ?? ""
const label = `${labelText}${punctuation} ${buttonText}`

return <Button ref={anchorRef} aria-label={label} {...inputProps} />
```

Reply:
Haha I see this is exactly what you did 😄

Reply:
Yeah ended up going with your suggestion from this comment: https://github.com/primer/react/pull/4389#issuecomment-1998145910 😄